lib/settings/settings_screen.dart

```diff
@@ -39,37 +47,195 @@ class SettingsScreen extends StatefulWidget {
 
 class SettingsScreenState extends State<SettingsScreen> {
   int _selectedIndex = 0;
+  OrionConfig config = OrionConfig();
+  Logger logger = Logger('Settings');
+  late bool needsRestart;
+  final GlobalKey<WifiScreenState> _wifiScreenKey =
+      GlobalKey<WifiScreenState>();
+  final ValueNotifier<bool> isConnected = ValueNotifier(false);
+  late Future<void> _wifiScreenFuture;
+
+  @override
+  void initState() {
+    super.initState();
+    needsRestart = config.getFlag('needsRestart', category: 'internal');
+    _wifiScreenFuture = _initializeWifiScreen();
+  }
+
+  Future<void> _initializeWifiScreen() async {
+    final bool initialConnectionStatus = await _checkInitialConnectionStatus();
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      if (mounted) {
+        isConnected.value = initialConnectionStatus;
+      }
+    });
+  }
+
+  Future<bool> _checkInitialConnectionStatus() async {
+    try {
+      final result = await Process.run('ping', ['-c', '1', 'google.com']);
```

Comment:
I don't think this is an appropriate check for wifi connection status? Internet status, sure, but I might run my machine on an isolated network.

Reply:
Resolved by [f161435](https://github.com/TheContrappostoShop/Orion/pull/18/commits/f161435bb86c4bbea504e99a66269aff109012bc).